atef/config.py

```diff
@@ -53,8 +41,29 @@ class Configuration:
     description: Optional[str] = None
     #: Tags tied to this configuration.
     tags: Optional[List[str]] = None
-    #: Comparison checklist for this configuration.
-    checklist: List[IdentifierAndComparison] = field(default_factory=list)
+
+
+@dataclass
+class ConfigurationGroup(Configuration):
+    """
+    Configuration group.
+    """
+    #: Configurations underneath this group.
+    configs: List[Configuration] = field(default_factory=list)
+    #: Values that can be reused in comparisons underneath this group.
+    values: Dict[str, Any] = field(default_factory=dict)
```

Comment:
From the verbose PR text:
> Adds a conceptual values dictionary that can be configured at a Group level, which may be reused in dynamic device configurations (all a TODO)

I'm thinking that you could define values at the configuration group level and reuse them in tests underneath. It's more a proposal of something I thought would be useful than anything else. If you think it'd be useful, let's leave it in; if not, let's nuke it before this PR goes through.

Reply:
I have no idea if it'll be useful or not- maybe I just need to see a specific use case. I re-read the verbose PR text before I made the above comment.

Is this basically setting a global variable for a group configuration? e.g.
```
full_vacuum = 1e-10
...
vgc_02.pressure < full_vacuum
```
I could see it getting some use if it's something like that

Reply:
Correct. The idea would be that you would only have to change the value in one place instead of (e.g) 10.

The scoping rules would be such that the parent group values have precedence over the grandparent group values (and any ancestor groups). Values could come from any ancestor if undefined on the parent group.

That means you could have a set of values defined in the root of the configuration that get reused throughout